### ISSUE TITLE
Update 1.23.1 -> 1.23.2 Failed on FreeNAS 

### ISSUE THREAD

**Issue by abrbon** [netdata/netdata#9563]
Hi,

Just tried to update my Netdata from 1.23.1 -> 1.23.2 and got the following error:

```
aclk/aclk_stats.o: In function `aclk_stats_main_thread':
aclk_stats.c:(.text+0x1b1): undefined reference to `mempcpy'
cc: error: linker command failed with exit code 1 (use -v to see invocation)
gmake[2]: *** [Makefile:4327: netdata] Error 1
gmake[2]: Leaving directory '/root/netdata-v1.23.2'
gmake[1]: *** [Makefile:5460: all-recursive] Error 1
gmake[1]: Leaving directory '/root/netdata-v1.23.2'
gmake: *** [Makefile:2947: all] Error 2
 FAILED
```   

This is on my TrueNAS 12 installed in a jail (iocage).

```
FreeBSD netdata-custom 12.1-STABLE FreeBSD 12.1-STABLE 3ef17902a90(HEAD) TRUENAS  amd64
1201517
```
Also tried to download the tarball for 1.23.2 and tried to install that but exact same error.

**Comment by prologic** [comment]
Hi @abrbon I'm also working on this in other related/similar issues on FreeNAS 11.3. Unfortunately I have little experience with FreeBSD systems myself so I'm having to learn about how to use `iocage` and try to figure out what's wrong with our build system on FreeNAS/TrueNAS systems. I _tried_ the hard way to build straight up on a FreeNAS box (_no jail_) only to find that the compiler and build tools are a bit broken or crippled for some reason.

**Comment by prologic** [comment]
If you are able to help with this in any way please let me know or free to put up a PR or two :)

**Comment by abrbon** [comment]
> Hi @abrbon I'm also working on this in other related/similar issues on FreeNAS 11.3. Unfortunately I have little experience with FreeBSD systems myself so I'm having to learn about how to use `iocage` and try to figure out what's wrong with our build system on FreeNAS/TrueNAS systems. I _tried_ the hard way to build straight up on a FreeNAS box (_no jail_) only to find that the compiler and build tools are a bit broken or crippled for some reason.

Hi @prologic ,
Coming more from a Linux background i also do not have a lot of experience with FreeBSD but setting up a jail (iocage) with FreeNAS is pretty straightforward and not to complicated.
You should never install or build anything directly in FreeNAS but always use a jail. That is best practice.
Inside the jail you can install the basic packages needed for building from tarball.

Unfortunately i do not have a lot of time to 'dig in' to this but i can try to help testing things or anything like that.
Btw, what do you mean with "PR" ?

**Comment by abrbon** [comment]
> If you are able to help with this in any way please let me know or free to put up a PR or two :)

Haven't written any (real) code since the 90's back in the day of Delphi :)
So not sure if i can help you with putting up that PR :D

**Comment by abrbon** [comment]
Good news!
Fixed the issue.

If you run the 11.3 Release of your jail (either from FreeNAS 11.3 or the new TrueNAS 12.1) you need to change the following:

In the aclk_stats.c file in the aclk\ folder you need to change the "mempcpy" command to "memcpy".

After this is could build the tarball without any issue.

**Comment by prologic** [comment]
"Delphi' pfft :) Good 'ol days!

**Comment by prologic** [comment]
> Fixed the issue for the 1.23.2 tarball

Thank you! I'll try to repeat your steps and come up with a patch to fix this properly for all users :)
